lib/Target/OpenFhePke/OpenFhePkeEmitter.h

```diff
@@ -28,14 +35,31 @@ namespace heir {
 namespace openfhe {
 
 /// Translates the given operation to OpenFhePke.
-::mlir::LogicalResult translateToOpenFhePke(
-    ::mlir::Operation *op, llvm::raw_ostream &os,
-    const OpenfheImportType &importType);
+::mlir::LogicalResult translateToOpenFhePke(::mlir::Operation *op,
+                                            llvm::raw_ostream &os,
+                                            const OpenfheImportType &importType,
+                                            const std::string &weightsFile);
+
+struct Weights {
+  std::map<std::string, std::vector<float>> floats;
```

Comment:
Could you add some comments about the meaning of these fields? At first glance I'd guess the key is an SSA value name, but it takes a bit of searching to confirm. And the values are always dense elements attr values, but is a non-1D shape supported? (I think not, or else the shape would be lost in the archive and I don't see any code to reshape)

Reply:
done